txnsync/txngroups.go

```diff
@@ -157,18 +155,12 @@ func decodeTransactionGroups(data []byte, compressionFormat byte, genesisID stri
 	return txnGroups, nil
 }
 
-func decompressTransactionGroupsBytes(data []byte) (decoded []byte, err error) {
-	zr, err := gzip.NewReader(bytes.NewBuffer(data))
-	defer zr.Close()
-	if err != nil {
-		return nil, fmt.Errorf("error gzip decompressing data: %w", err)
-	}
-	var buf bytes.Buffer
-	_, err = buf.ReadFrom(zr)
-	if err != nil {
-		return nil, fmt.Errorf("error gzip decompressing data: %w", err)
+func decompressTransactionGroupsBytes(data []byte, lenDecompressedBytes uint64) (decoded []byte, err error) {
+	if lenDecompressedBytes > maxEncodedTransactionGroupBytes || lenDecompressedBytes < uint64(len(data)) {
```

Comment:
one more restriction : make sure that the expected decompression ratio makes sense -
i.e. ratio of 0% - 95% of compression gains might be reasonable. Anything better is unlikely.